src/librustc_target/spec/thumbv8m_none_eabi.rs

```diff
@@ -0,0 +1,33 @@
+// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// Targets the Cortex-M23 processor (Baseline ARMv8-M)
+
+use spec::{LinkerFlavor, LldFlavor, Target, TargetOptions, TargetResult};
+
+pub fn target() -> TargetResult {
+    Ok(Target {
+        llvm_target: "thumbv8m.base-none-eabi".to_string(),
+        target_endian: "little".to_string(),
+        target_pointer_width: "32".to_string(),
+        target_c_int_width: "32".to_string(),
+        data_layout: "e-m:e-p:32:32-i64:64-v128:64:128-a:0:32-n32-S64".to_string(),
+        arch: "arm".to_string(),
+        target_os: "none".to_string(),
+        target_env: String::new(),
+        target_vendor: String::new(),
+        linker_flavor: LinkerFlavor::Lld(LldFlavor::Ld),
+
+        options: TargetOptions {
```

Comment:
The baseline architecture doesn't support unaligned access, so `features: "+strict-align".to_string(),` will need to be added here as I believe it's not enabled by default.

Reply:
So, it looks like the "Main Extension" from 8-M was a baseline in 7-M? And this newly added target (used for Cortex-M23) is not really a baseline?

Any other notable changes in 8-M affecting codegen? The "changelog" in ARM ARM doesn't tell much to me.

Reply:
Aha! Good catch @parched, I will adjust.

I took a look at this description of changes here: https://community.arm.com/processors/trustzone-for-armv8-m/b/blog/posts/five-key-features-of-the-arm-cortex-m23-processor?CommentId=f95326ae-7e58-4829-9874-a1ebd4a141e9

I think I may have atomic instruction support incorrectly configured. Per the above blog post:
>  In addition, to provide atomic support for C11/C++11, the load-acquire and store-release instructions are included from ARMv8-A

There is no mention of atomic CAS support, I think that I may need to set `atomic_cas: false` while keeping `max_atomic_width: Some(32),`. https://reviews.llvm.org/D15283?id=42050 also seems to hint at this. Does this seem correct?